### DISCUSSION TITLE
微调中文PPOCR-v4出错，输入数据维度不匹配

### DISCUSSION THREAD

**Discussion by 0-Maxwei-0** [PaddlePaddle/PaddleOCR#12810] [Q&A]
return self.forward(*inputs, **kwargs)
  File "/root/miniconda3/lib/python3.8/site-packages/paddle/nn/layer/conv.py", line 712, in forward
    out = F.conv._conv_nd(
  File "/root/miniconda3/lib/python3.8/site-packages/paddle/nn/functional/conv.py", line 140, in _conv_nd
    pre_bias = _C_ops.conv2d(
ValueError: (InvalidArgument) The input of Op(Conv) should be a 4-D or 5-D Tensor. But received: input's dimension is 3, input's shape is [64, 240, 256].
  [Hint: Expected in_dims.size() == 4 || in_dims.size() == 5 == true, but received in_dims.size() == 4 || in_dims.size() == 5:0 != true:1.] (at /paddle/paddle/phi/infermeta/binary.cc:470)

**Comment by 0-Maxwei-0** [comment]
同一批数据，在v3上调是没问题的，v4上就会出现这样的错误，不知道是什么原因呢？谢谢

**Comment by IbrarBabar009** [comment]
Please share your yml file

and what is the version you are using of PaddleOCR, I am asking about the version of github PaddleOCR you are using?

share a more output please

**Comment by 0-Maxwei-0** [comment]
> Please share your yml file
> 
> and what is the version you are using of PaddleOCR, I am asking about the version of github PaddleOCR you are using?
> 
> share a more output please

Global:
  debug: false
  use_gpu: true
  epoch_num: 1000
  log_smooth_window: 20
  print_batch_step: 20
  save_model_dir: ./output/rec_dkd_400w_svtr_ctc_lcnet_blank_dkd0.1/
  save_epoch_step: 40
  eval_batch_step:
  - 0
  - 2000
  cal_metric_during_train: true
  pretrained_model: /root/autodl-tmp/ch_PP-OCRv4_rec_train/student.pdparams
  checkpoints: # ./output/rec_dkd_400w_svtr_ctc_lcnet_blank_dkd0.1/latest
  save_inference_dir: null
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  max_text_length: &max_text_length 40
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ppocrv3.txt
Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.0001
    warmup_epoch: 2
  regularizer:
    name: L2
    factor: 3.0e-05
Architecture:
  model_type: rec
  name: DistillationModel
  algorithm: Distillation
  Models:
    Teacher:
      pretrained: 
      freeze_params: true
      return_all_feats: true
      model_type: rec
      algorithm: SVTR
      Transform: null
      Backbone:
        name: SVTRNet
        img_size:
        - 48
        - 320
        out_char_num: 40
        out_channels: 192
        patch_merging: Conv
        embed_dim:
        - 64
        - 128
        - 256
        depth:
        - 3
        - 6
        - 3
        num_heads:
        - 2
        - 4
        - 8
        mixer:
        - Conv
        - Conv
        - Conv
        - Conv
        - Conv
        - Conv
        - Global
        - Global
        - Global
        - Global
        - Global
        - Global
        local_mixer:
        - - 5
          - 5
        - - 5
          - 5
        - - 5
          - 5
        last_stage: false
        prenorm: true
      Head:
        name: MultiHead
        head_list:
          - CTCHead:
              Neck:
                name: svtr
                dims: 120
                depth: 2
                hidden_dims: 120
                kernel_size: [1, 3]
                use_guide: True
              Head:
                fc_decay: 0.00001
          - NRTRHead:
              nrtr_dim: 384
              max_text_length: *max_text_length
    Student:
      pretrained: 
      freeze_params: false
      return_all_feats: true
      model_type: rec
      algorithm: SVTR
      Transform: null
      Backbone:
        name: PPLCNetV3
        scale: 0.95
      Head:
        name: MultiHead
        head_list:
          - CTCHead:
              Neck:
                name: svtr
                dims: 120
                depth: 2
                hidden_dims: 120
                kernel_size: [1, 3]
                use_guide: True
              Head:
                fc_decay: 0.00001
          - NRTRHead:
              nrtr_dim: 384
              max_text_length: *max_text_length
Loss:
  name: CombinedLoss
  loss_config_list:
  - DistillationDKDLoss:
      weight: 0.1
      model_name_pairs:
      - - Student
        - Teacher
      key: head_out
      multi_head: true
      alpha: 1.0
      beta: 2.0
      dis_head: gtc
      name: dkd
  - DistillationCTCLoss:
      weight: 1.0
      model_name_list:
      - Student
      key: head_out
      multi_head: true
  - DistillationNRTRLoss:
      weight: 1.0
      smoothing: false
      model_name_list:
      - Student
      key: head_out
      multi_head: true
  - DistillCTCLogits:
      weight: 1.0
      reduction: mean
      model_name_pairs:
      - - Student
        - Teacher
      key: head_out
PostProcess:
  name: DistillationCTCLabelDecode
  model_name:
  - Student
  key: head_out
  multi_head: true
Metric:
  name: DistillationMetric
  base_metric_name: RecMetric
  main_indicator: acc
  key: Student
  ignore_space: false
Train:
  dataset:
    name: SimpleDataSet
    data_dir: /root/autodl-tmp/rec/train
    label_file_list:
    - /root/autodl-tmp/rec/train.txt
    ratio_list:
    - 1.0
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecAug:
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  loader:
    shuffle: true
    batch_size_per_card: 64
    drop_last: true
    num_workers: 1
    use_shared_memory: true
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: /root/autodl-tmp/rec/val
    label_file_list:
    - /root/autodl-tmp/rec/val.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 64
    num_workers: 1
profiler_options: null



使用paddleOCR 2.6.1

**Comment by 0-Maxwei-0** [comment]
我参考了这篇回答  [#10700](https://github.com/PaddlePaddle/PaddleOCR/issues/10700#issue-1861611688)进行代码的修改。

**Comment by tohsakask** [comment]
不调整neck就行了，如果是SVTR还是用的reshape

**Comment by minakami443** [comment]
你用的config是Teacher-Student蒸餾用的, 會報錯正常
要fine-tune的話參照"\PaddleOCR\configs\rec\PP-OCRv4\ch_PP-OCRv4_rec.yml"去改

**Comment by iPengXPro** [comment]
请问v4模型微调您跑通了吗？能分指点一下如何操作吗？
